hdfs-dataset-sink/src/test/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkConfigurationTests.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.dataset.sink;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.data.hadoop.fs.FsShell;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = DatasetSinkApplication.class)
+@WebIntegrationTest({"server.port:0",
+		"spring.hadoop.fsUri=file:///",
+		"directory=${java.io.tmpdir}/dataset",
+		"namespace=test",
+		"batchSize=2",
+		"idleTimeout=2000"})
+@DirtiesContext
+public class DatasetSinkConfigurationTests {
+
+	@Autowired
+	ConfigurableApplicationContext applicationContext;
+
+	@Value("${directory}/${namespace}")
+	private String testDir;
+
+	@Autowired
+	private FsShell fsShell;
+
+	@Autowired
+	@Bindings(DatasetSink.class)
+	private Sink sink;
+
+	@Before
+	public void setup() {
+		if (fsShell.test(testDir)) {
+			fsShell.rmr(testDir);
+		}
+	}
+
+	@Test
+	public void testWritingSomething() throws IOException, InterruptedException {
+		sink.input().send(new GenericMessage<>("Foo"));
+		sink.input().send(new GenericMessage<>("Bar"));
+	}
+
+	@After
+	public void checkFilesClosedOK() throws IOException, InterruptedException {
+		applicationContext.close();
```

Comment:
What is the reason to `close()` ctx manually if you are based on the Spring Test Framework?
Especially it will be very dangerous when you add one more `@Test`  to this class